### PR TITLE
Refactor SortMeRNA

### PR DESCRIPTION
**Pull request by bebatut**
Hi,

With this PR:
- Supports paired-end data
- Supports OTU map option
- Fix issue of expected BLAST, SAM or OTU file if best/first alignment option is checked
- Change SAM output to BAM output

Bérénice

### CODE REVIEW

File: tools/rna_tools/sortmerna/sortmerna.xml [hunk 1]
```diff
@@ -46,93 +47,117 @@ sortmerna --version 2>&1|grep 'SortMeRNA version'
...
         --match $match
         --mismatch $mismatch
         --gap_open $gap_open
         --gap_ext $gap_ext
         -N $ambiguous_letter
-    	-a \${GALAXY_SLOTS:-1}
+        -a \${GALAXY_SLOTS:-1}
+    #if $report.report_type != 'None'
+        &&
+        samtools view -S -b aligned.sam > $output_bam
```
Comment (bgruening) [inline tools/rna_tools/sortmerna/sortmerna.xml]: output_bam -> quotation

Comment (bebatut) [reply tools/rna_tools/sortmerna/sortmerna.xml]: Updated

**Comment by bebatut** [comment]
I changed the tests for the OTU part

**Comment by bgruening** [comment]
Urgs, seems that we need to sort the BAM file.

**Comment by bgruening** [comment]
ok, the BAM file needs to be updated, then it should be ready to go! :)
Thanks B!

**Pull request merged**